src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/AceEditorFocusTracker.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * AceEditor.java
+ *
+ * Copyright (C) 2009-16 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+package org.rstudio.studio.client.workbench.views.source.editors.text;
+
+import com.google.inject.Singleton;
+
+@Singleton
+public class AceEditorFocusTracker
```

Comment:
I'm imagining that this might be used across more modules than just `Source` in the future so I think it's okay to have it separate.